include/xwidgets/xnumber.hpp

```diff
@@ -87,8 +89,21 @@ namespace xw
     template <class D>
     inline void xnumber<D>::set_defaults()
     {
-        this->_model_name() = "BoundedFloatModel";
-        this->_view_name() = "BoundedFloatView";
```

Comment:
This was not a valid frontend model. It's (likely) `BoundedFloatTextModel` (as fixed in this PR).

Reply:
Indeed, `BoundedFloatModel` and `BoundedFloatView` do not exist. Though I'm not sure `BoundedFloatTextModel` makes more sense.

`xnumber` is supposed to be a base class that's not used directly. It could very well be a virtual class and subclasses would be required to provide a `_model_name` and `_view_name`.

Maybe you could simply remove those two lines from the `set_default`.

Reply:
> xnumber is supposed to be a base class that's not used directly. It could very well be a virtual class and subclasses would be required to provide a _model_name and _view_name.

That is what I noticed. I'm wondering since there is a frontend model if this would make sense to also make it a valid widget with a `BoundedFloatTextModel` ?

There are also more things wring with `number`: it is bounded, so everyone that inherits from `xnumber` gets a `min` and a `max` even if the frontend model does not manage it.

My proposal (WIP) is that `xnumber` inherits from `xnumeral` (unbounded).
And to change these confusing names along the way to, say, `number` (unbounded) and `bounded_number`/`interval` (bounded).